jwst/master_background/master_background_step.py

```diff
@@ -236,29 +236,48 @@ def copy_background_to_surf_bright(spectrum):
         spec.spec_table['BKGD_ERROR'][:] = 0.
 
 
-def split_container(container):
-    """Divide a ModelContainer with science and background into one of each
+def split_container(container: ModelContainer, fields=['science', 'background']):
```

Comment:
Should we move this function to a library somewhere, since it's now needed in multiple places?

Reply:
I am all for that, but I wonder where it should go? How about with ModelContainer inside datamodels.container?

Reply:
That sounds appropriate to me.

Reply:
A few words of caution here:
- split_container is also used in [calwebb_spec3](https://github.com/spacetelescope/jwst/blob/d6435f18465155fa32fecb8c4a94ab192dcec1d0/jwst/pipeline/calwebb_spec3.py#L12)
- the current function only works for a `ModelContainer` read from an association (or a "derivative" of one). A container made with a list of models won't have for instance `asn_pool_name`
- the current function attempts to keep the "member" list updated (for the "science" container) but won't produce expected results if a models `meta.asn.exptype` is changed (the split happens based on the model metadata but the member split happens based on the association values)

If I'm following the changes in this PR, only the first science model is used (so the science container is discarded after it's split). This discards the association information and in this case a list of models grouped by `meta.asn.exptype` seems easier to follow (and less prone to the issues above) vs using `split_container`.

Reply:
@braingram ok I hear you RE the `split_container` function only operating on a ModelContainer that is read from an association.  But it also feels a bit strange to define a new function in this step that, well, splits a container, even if the two operate slightly differently.  I guess philosophically I would expect a function to exist that will split a container by exposure type, and have it work on any valid ModelContainer object.  So I guess for now, because I know ModelContainer is soon being deprecated anyway, I'll write the separate function, but something to keep in mind perhaps for ModelLibrary.

With regards to being used also in calwebb_spec3, yes, that's true, and the proposed changes are fully back-compatible with the old behavior.  But that point is moot if I make your suggested fix and bypass split_container entirely.